### PR TITLE
snap,store: switch to the new snaps/metadata endpoint, introduce and start capturing DeveloperID

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

File: store/store_test.go [hunk 1]
```diff
@@ -756,24 +766,43 @@ func (t *remoteRepoTestSuite) TestUbuntuStoreFindAuthFailed(c *C) {
...
             {
                 "_links": {
                     "self": {
-                        "href": "https://search.apps.staging.ubuntu.com/api/v1/package/%[1]s"
+                        "href": "https://search.apps.ubuntu.com/api/v1/package/buPKUD3TKqCOgLEjjHx5kSiCpIs5cMuQ"
                     }
                 },
-                "download_url": "https://public.apps.staging.ubuntu.com/download-snap/%[1].snap",
+                "anon_download_url": "https://public.apps.ubuntu.com/anon/download-snap/buPKUD3TKqCOgLEjjHx5kSiCpIs5cMuQ_26.snap",
+                "architecture": [
```
Comment (mvo5) [inline store/store_test.go]: Is this `architecture` (singular) and a list :/ ?

**Comment by mvo5** [comment]
:+1:

File: store/store_test.go (hunk 1 above)
Comment (pedronis) [inline store/store_test.go]: afaict yes

**Comment by niemeyer** [comment]
What's the migration plan?

File: snap/info_test.go [hunk 2]
```diff
@@ -59,13 +59,15 @@ func (s *infoSuite) TestSideInfoOverrides(c *C) {
 		EditedDescription: "fixed desc",
 		Revision:          snap.R(1),
 		SnapID:            "snapidsnapidsnapidsnapidsnapidsn",
+		DeveloperID:       "deviddeviddeviddeviddeviddevidde",
```
Comment (niemeyer) [inline snap/info_test.go]: Is this a random string or a readable developer username?  The copy & paste from the line above might give an improper feeling.

**Comment by niemeyer** [comment]
Looks fine. Just wondering about the migration plan indeed. Having it unused and with dup data might make migration harder than having ana atomic change that moves over. The atomic change must happen anyway, as we will never be sure if everyone now has the data.

**Pull request merged**